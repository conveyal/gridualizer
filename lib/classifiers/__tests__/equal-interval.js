/* global describe, it, expect */

import {scaleLog} from 'd3-scale'
import range from 'lodash/range'

import equalInterval from '../equal-interval'

const mockGrid = {
  min: 0,
  max: 1000,
  data: range(0, 1001)
}

describe('classifiers > equal-interval', () => {
  it('should handle a linear classifier', () => {
    expect(equalInterval()(mockGrid, 5)).toMatchSnapshot()
  })

  it('should handle a nonlinear classifier', () => {
    const scale = scaleLog()
      .domain([1, 1000])
      .clamp(true)

    expect(equalInterval({ scale })(mockGrid, 5)).toMatchSnapshot()
  })
})
