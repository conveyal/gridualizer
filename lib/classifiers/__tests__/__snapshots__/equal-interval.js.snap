// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classifiers > equal-interval should handle a linear classifier 1`] = `
Array [
  200,
  400,
  600,
  800,
  1000,
]
`;

exports[`classifiers > equal-interval should handle a nonlinear classifier 1`] = `
Array [
  3.9810717055349727,
  15.848931924611136,
  63.095734448019314,
  251.1886431509581,
  1000,
]
`;
